src/main/java/com/varukha/onlinebookstore/service/order/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+        order.setStatus(Order.Status.PENDING);
+        order.setTotal(calculateTotalCartItemsPrice(authenticatedUser.getId()));
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(orderRequestDto.getShippingAddress());
+        order.setOrderItems(orderItems);
+        Order savedOrder = orderRepository.save(order);
+        orderItems.forEach(orderItem -> orderItem.setOrder(savedOrder));
+        orderItemRepository.saveAll(orderItems);
+        return savedOrder;
+    }
+
+    @Override
+    public List<OrderDto> getAll(Pageable pageable) {
+        return orderRepository.findAllOrders(pageable)
+                .stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public Set<OrderItemDto> getAllOrderItemsByOrderId(Long id) {
+        return orderRepository.findOrderById(id)
+                .map(orderItem -> orderItem.getOrderItems()
+                        .stream()
+                        .map(orderItemsMapper::toDto)
+                        .collect(Collectors.toSet()))
+                .orElse(Collections.emptySet());
+    }
+
+    @Override
+    public OrderItemDto getOrderItemFromOrderById(Long orderId, Long itemId) {
+        return orderItemRepository.findOrderItemByIdAndOrderId(itemId, orderId)
+                .map(orderItemsMapper::toDto)
+                .orElseThrow();
+    }
+
+    @Override
+    public OrderDto updateOrderStatus(Long id, UpdateOrderStatusRequestDto statusRequestDto) {
+        return orderRepository.findById(id)
+                .map(order -> {
+                    order.setStatus(statusRequestDto.getStatus());
+                    orderRepository.save(order);
+                    return orderMapper.toDto(order);
+                })
+                .orElseThrow(() -> new EntityNotFoundException("Order "
+                        + "not found by id: " + id));
+    }
+
+    private BigDecimal calculateTotalCartItemsPrice(Long userId) {
+        return shoppingCartRepository.findByUserId(userId)
+                .map(shoppingCart -> shoppingCart.getCartItems()
+                        .stream()
+                        .map(cartItem -> cartItem.getBook().getPrice()
+                                .multiply(BigDecimal.valueOf(cartItem.getQuantity())))
+                        .reduce(BigDecimal.ZERO, BigDecimal::add))
+                .orElse(BigDecimal.ZERO);
+    }
+
+    private Set<OrderItem> getOrderItems(Long userId) {
+        return shoppingCartRepository.findByUserId(userId)
+                .map(shoppingCart -> shoppingCart.getCartItems()
+                        .stream()
+                        .map(orderItemsMapper::toModel)
+                        .collect(Collectors.toSet()))
+                .orElse(Collections.emptySet());
+    }
+
+    @Transactional
+    public void clearShoppingCart(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
```

Comment:
Why order service is responsible for interaction with shopping cart? Dont you have shopping cart service for it?